src/rbd.cc

```diff
@@ -2470,6 +2469,34 @@ static int do_show_status(librados::IoCtx &io_ctx, librbd::Image &image,
   return 0;
 }
 
+static int do_rename(librbd::RBD &rbd, librados::IoCtx& io_ctx,
+		     librbd::Image &image, const char *imgname, const char *destname)
+{
+  int r = rbd.open_read_only(io_ctx, image, imgname, NULL);
+  if (r < 0) {
+    cerr << "rbd: error opening image " << imgname << ": "
+         << cpp_strerror(-r) << std::endl;
+    return -r;
+  }
+  bool watch_exists = false;
+  r = do_show_status(io_ctx, image, imgname, NULL, &watch_exists);
```

Comment:
moving the listwatchers() part of do_show_status() into a helper that this and do_show_status() use would be clearer